docs/using-the-compiler.rst

```diff
@@ -273,3 +274,16 @@ Output Description
         }
       }
     }
+
+
+Error types
+~~~~~~~~~~~
+
+1. ``JSONError``: JSON input doesn't conform to the required format, e.g. input is not a JSON object, the language is not supported etc.
+2. ``IOError``: Input/Output errors, e.g. invalid URLs
+3. ``InternalCompilerError``: internal compiler error, e.g. compilation failed, JSON logic error, JSON runtime error etc.
```

Comment:
JSON errors are not reported as part of this.

Reply:
the information comes from code:

```
./libsolidity/interface/StandardCompiler.cpp:457:		return formatFatalError("InternalCompilerError", string("JSON logic exception: ") + _exception.what());
./libsolidity/interface/StandardCompiler.cpp:461:		return formatFatalError("InternalCompilerError", string("JSON runtime exception: ") + _exception.what());
```

Reply:
Ah ok, those messages are a bit misleading. It still means that the compiler code is broken, not that the supplied data is broken.

All it should say is something like "internal bug in the compiler (should be reported as an issue)".

Reply:
I see, thanks a lot